File: 프로그래머스/2/42885. 구명보트/구명보트.py
from collections import deque

def solution(people, limit):
    people.sort()
    q = deque(people)
    answer = 0
    cur_limit = 0
    while len(q)>1:
        if q[-1] + q[0] <= limit:
            q.popleft()
            q.pop()
            answer += 1
        else:
            q.pop()
            answer += 1
    if q:
        answer += 1
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream> 

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    int i = 0, j = people.size()-1;
    sort(people.begin(),people.end());
    while (i <= j){
        if (people[i] + people[j] <= limit) {
            i++; 
        }
        j--;
        answer++;
    }
    return answer;
    
}